pkg/virt-handler/vm.go

```diff
@@ -2517,6 +2517,10 @@ func (d *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
 			return fmt.Errorf("failed to adjust resources: %v", err)
 		}
 	} else if vmi.IsRunning() {
+		if err := d.hotplugSriovInterfaces(vmi); err != nil {
+			log.Log.Object(vmi).Error(err.Error())
```

Comment:
Shouldn't you re-enqueue the vmi is such case? How do you make sure that the vmi reconcile will be invoked again?

Also, on our hangouts meeting @EdDev mentioned the `d.hotplugSriovInterfaces` is async. In such case even if an error is not returned, the operation may fail. How do we make sure the vmi reconcile in called again?

Reply:
So you're saying the `re-enqueqe` we do in the controller is redundant?
The reconcile for the vmi is periodically invoked anyway?

Reply:
> EDIT: or what you're saying is that in the sriov hotplug case the re-enqueue is not needed since the vm is running and for running vm we have periodic reconcile?

Yes.
Regarding the periodic VMI sync here are the references to what I described earlier:
The domains informer on virt-handler, triggers a VMI sync every 5 minutes [[1]](https://github.com/kubevirt/kubevirt/blob/a6272e9c8c728d051cdc01028aed994a84a5049e/cmd/virt-handler/virt-handler.go#L109) [[2]](https://github.com/kubevirt/kubevirt/blob/a6272e9c8c728d051cdc01028aed994a84a5049e/cmd/virt-handler/virt-handler.go#L251) [[3]](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/cache/cache.go#L339) [[4]](https://github.com/kubevirt/kubevirt/blob/a6272e9c8c728d051cdc01028aed994a84a5049e/pkg/virt-handler/cache/cache.go#L407) 
Other than that, virt-launcher domain-notifier client triggers a VMI sync (by sending domain modified event) every two minutes when QEMU guest-agent presents [[5]](https://github.com/kubevirt/kubevirt/blob/a6272e9c8c728d051cdc01028aed994a84a5049e/cmd/virt-launcher/virt-launcher.go#L355) [[6]](https://github.com/kubevirt/kubevirt/blob/a6272e9c8c728d051cdc01028aed994a84a5049e/cmd/virt-launcher/virt-launcher.go#L477) [[7]](https://github.com/kubevirt/kubevirt/blob/a6272e9c8c728d051cdc01028aed994a84a5049e/cmd/virt-launcher/virt-launcher.go#L157) [[8]](https://github.com/kubevirt/kubevirt/blob/a6272e9c8c728d051cdc01028aed994a84a5049e/pkg/virt-launcher/notify-client/client.go#L391-L396) [[9]](https://github.com/kubevirt/kubevirt/blob/a6272e9c8c728d051cdc01028aed994a84a5049e/pkg/virt-launcher/notify-client/client.go#L416) [[10]](https://github.com/kubevirt/kubevirt/blob/a6272e9c8c728d051cdc01028aed994a84a5049e/pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go#L341-L343)